lib/src/Base/Optim/HiGHS.cxx

```diff
@@ -0,0 +1,235 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief HiGHS linear solver
+ *
+ *  Copyright 2005-2025 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/HiGHS.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/OTconfig.hxx"
+
+#ifdef OPENTURNS_HAVE_HIGHS
+#include "Highs.h"
+#endif
+
+#include <chrono>
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(HiGHS)
+
+static const Factory<HiGHS> Factory_HiGHS;
+
+/* Constructor with no parameters */
+HiGHS::HiGHS()
+  : OptimizationAlgorithmImplementation()
+{
+  // Nothing to do here
+}
+
+
+/* Constructor that sets starting sample */
+HiGHS::HiGHS(const OptimizationProblem & problem)
+  : OptimizationAlgorithmImplementation(problem)
+
+{
+}
+
+
+/* Check whether this problem can be solved by this solver */
+// Inherited method: never used by HiGHS because solver_ calls its own checkProblem method.
+void HiGHS::checkProblem(const OptimizationProblem & problem) const
+{
+  // Cannot solve multi-objective problems
+  if (problem.hasMultipleObjective()) throw InvalidArgumentException(HERE) << "Bonmin does not support multi-objective optimization";
+
+  // No LeastSquaresProblem / NearestPointProblem
+  if (problem.hasResidualFunction() || problem.hasLevelFunction())
+    throw InvalidArgumentException(HERE) << "HiGHS does not support least squares or nearest point problems";
+
+  if (problem.hasEqualityConstraint())
+    throw InvalidArgumentException(HERE) << "HiGHS does not support equality constraints";
```

Comment:
A way to support it is to mimic Cobyla: f(x)=0 is replaced by f(x)>=-eps and f(x)<=eps for a small eps